lib/Backend/FlowGraph.cpp

```diff
@@ -4646,7 +4646,18 @@ BasicBlock::CheckLegalityAndFoldPathDepBranches(GlobOpt* globOpt)
             StackSym* src1Sym = instr->GetSrc1()->GetStackSym();
             FindValueInLocalThenGlobalValueTableAndUpdate(globOpt, localSymToValueMap, instr, instr->GetDst()->GetSym(), src1Sym);
         }
-        else if (!instr->GetSrc1()->IsIntConstOpnd())
+        else if (instr->GetSrc1()->IsIntConstOpnd())
+        {
+            Value **localValue = localSymToValueMap->FindOrInsertNew(instr->GetDst()->GetSym());
+            *localValue = globOpt->NewValue(IntConstantValueInfo::New(globOpt->alloc, instr->GetSrc1()->AsIntConstOpnd()->AsInt32()));
```

Comment:
Instead of creating a new value, can we use the existing value for the int constant? (GlobOpt::GetIntConstantValue)

Reply:
Yes, thank you.